pkg/storage/tsdb/index_cache.go

```diff
@@ -119,3 +125,17 @@ func newMemcachedIndexCache(cfg cache.MemcachedConfig, logger log.Logger, regist
 
 	return indexcache.NewTracingIndexCache(c, logger), nil
 }
+
+func newRedisIndexCache(cfg cache.RedisClientConfig, logger log.Logger, registerer prometheus.Registerer) (indexcache.IndexCache, error) {
+	client, err := cache.NewRedisClient(logger, "index-cache", cfg, prometheus.WrapRegistererWithPrefix("thanos_", registerer))
```

Comment:
We had a discussion [to deprecate `thanos_` metrics prefix](https://github.com/grafana/mimir/issues/4241). Should we just use `cortex_` prefix for the new change? Applicable also for memcached client metrics prefix few lines above.

Reply:
I'd rather make the change to all cache metrics at the same time. It's the same amount of work if we support `cortex_` for Redis metrics and `thanos_` for Memcached vs using `thanos_` for both now and `cortex_` for both later.

Reply:
To be more specific, using `thanos_` here now means Redis will work with our dashboards as currently written with no extra effort. Then, they can be migrated to use the `cortex_` prefix at the same time as the Memcached backend also with no extra effort.